pandas/core/series.py

```diff
@@ -2861,6 +2864,8 @@ def create_from_value(value, index, dtype):
 
         if is_datetimetz(dtype):
```

Comment:
so now seems that we have tons of cases where doing:

```
if is_datetimetz: 
   ....
elif is_period.....
    ...

elif is_categorical...
    ...
```

ideally we should have functions for various things based on the dtype object itself.

e.g. something like:

```
try:
    subarr = dtype._create_array(......)
except:
    pass
```